### PR TITLE
Add test to controller

### PR DESCRIPTION
**Pull request by navidshariaty**
Here are the test cases added for the controller:
- no object should be created when the pause annotation is set on the route object
- no object should be created when the route is not in the Admitted state
- the creation of HTTPProxy without TLS set on the route, and everything is alright
- the creation of HTTPProxy without TLS set on the route, and the route has a custom load balancer algorithm
- the creation of HTTPProxy without TLS set on the route, and the route has Ratelimit enabled
- the creation of HTTPProxy without TLS set on the route, and the route has the IP whitelist enabled
- the creation of TCPProxy with passthrough TLS enabled on the route
- the creation of HTTPProxy with edge TLS enabled on the route

Also, I had to change some parts of the code to make it easier to test.

### CODE REVIEW

File: internal/controller/suite_test.go [hunk 1]
```diff
@@ -60,16 +74,59 @@ var _ = BeforeSuite(func() {
...
+	}()
+
 })
 
 var _ = AfterSuite(func() {
 	By("tearing down the test environment")
-	err := testEnv.Stop()
-	Expect(err).NotTo(HaveOccurred())
+	_ = testEnv.Stop()
+	Expect(testEnv.Stop()).To(Succeed())
```
Comment (therealak12) [inline internal/controller/suite_test.go:131]: Is it required to call `testEnv.Stop` twice?

File: internal/controller/route/handler.go [hunk 2]
```diff
@@ -555,6 +554,11 @@ func (r *Reconciler) assembleTLSSecret(route *routev1.Route) *corev1.Secret {
 // the routes are sorted in ascending order by .metadata.creationTimestamp field
 func (r *Reconciler) getSameHostRoutes(ctx context.Context, namespace, host string) ([]routev1.Route, error) {
 	sameHostRouteList := &routev1.RouteList{}
+
+	if err := r.List(ctx, sameHostRouteList, client.InNamespace(namespace)); err != nil {
```
Comment (therealak12) [inline internal/controller/route/handler.go:558]: Why are we adding this section?

File: internal/controller/controller_test.go [hunk 3]
```diff
@@ -0,0 +1,603 @@
...
+
+var (
+	err             error
+	ServiceWeight   int32 = 100
+	ServiceFooPorts       = []v12.ServicePort{{Name: "https", Port: 443}}
+	RouterTypeMeta        = v1.TypeMeta{Kind: KindRoute, APIVersion: APIVersionRoute}
+	ServiceTypeMeta       = v1.TypeMeta{Kind: KindService, APIVersion: APIVersionService}
+)
+
+var _ = Describe("Testing Route to HTTPProxy Controller", func() {
```
Comment (therealak12) [inline internal/controller/controller_test.go:55]: Adding a test case for creating an httpproxy for multiple routes with the same host and different paths could be helpful.

File: internal/controller/route/handler.go (hunk 2 above)
Comment (navidshariaty) [reply internal/controller/route/handler.go]: oh, I forgot to remove this section. I wanted to test sth. thanks for mentioning it. it's resolved now

**Pull request merged**